### ISSUE TITLE
Everything should be async

### ISSUE THREAD

**Issue by Xophmeister** [wtsi-hgi/irobot#9]
The API server is all asynchronous, but it interfaces with a bunch of things that aren't; they use threading and the listener pattern to achieve concurrency (modulo the GIL, of course). This, I believe, will cause horrible pausing issues when any async code has to wait on non-async code. For example, every current user's HTTP session could be locked up while one user requests checksums for a chunk that doesn't exist, or if their authentication handler can't connect to its external service.

The following definitely need to be converted to using async:
* The checksummer
* The authentication handlers

The iRODS interface should probably also be converted, but it's not a priority as it's "far enough away" from the API server that it probably won't matter.
